File: SOLID/single-responsibility.py
class Report:
    def __init__(self,data):
        self.data=data
    def generate(self):
        return f"report: {self.data}"

class FileSaver:
    def save(self,filename,content):
        with open(filename,"w") as f:
            f.write(content)
class EmailSender:
    def send(self,email,content):
        print(f"Sending '{content}' to {email}")
report=Report("Sales Data")
content=report.generate()
file=FileSaver()
file.save("report.txt",content)
mail=EmailSender()
mail.send("abc@gmail.com",content)

File: SOLID/SRP/single-responsibility.go
package main

import (
	"fmt"
	"os"
)

type Report struct{
	Data string
}
func(r Report)Generate()string{
	return fmt.Sprintf("Report %s",r.Data)
}
type FileSaver struct{

}
func( f FileSaver) Save(filename string,content string) error{
	return os.WriteFile(filename,[]byte(content),0644)
}
type EmailSender struct{}
func(e EmailSender)Send(email string,content string){
	fmt.Printf("Sending '%s' to %s\n",content,email)
}
func main() {

	report := Report{
		Data: "Sales Data In Go",
	} 
	content := report.Generate()
	file := FileSaver{}
	err := file.Save("report.txt", content)
	if err != nil {
		fmt.Println("Error saving file:", err)
		return
	}
	mail := EmailSender{}
	mail.Send("abc@gmail.com", content)

}